component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPKeyGenerator.java

```diff
@@ -94,7 +94,12 @@ private static Map<String, String> generateClaims(String username, boolean refre
                 if (StringUtils.isBlank(storedSecretKey) || refresh) {
                     TOTPAuthenticatorKey key = generateKey(tenantDomain, context);
                     generatedSecretKey = key.getKey().toCharArray();
-                    claims.put(secretKeyClaim, TOTPUtil.encrypt(String.valueOf(generatedSecretKey)));
+                    Map<String, String> claimProperties = TOTPUtil.getClaimProperties(tenantDomain,secretKeyClaim);
+                    if (claimProperties.get("EnableEncryption") != null) {
```

Comment:
Shall we add a constant reference to this ?

Reply:
This change will be reverted after the migration is completed